### PR TITLE
Feature login

### PR DESCRIPTION
**Pull request by fcarelse**
Got a visible login page

### CODE REVIEW

File: src/context/Auth/AuthContext.ts [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
+import {
+  LoginHandler,
+  AuthContextValue,
+  Credentials,
+} from "../../helper/Types";
 
 export const AuthContext = React.createContext({
   token: "",
-  login: () => {},
+  login: async ({ username, password }: Credentials) => ({ token: "" }),
```
Comment (Kivylius) [inline src/context/Auth/AuthContext.ts]: this can just be ReturnType<typeof function_here>['login']

File: src/pages/Login/LoginPage.tsx [hunk 2]
```diff
@@ -1,3 +1,111 @@
...
+    >
+      {"Copyright ©"}
+      <Link color="inherit" href="https://chatrooms.party/">
+        Todolist Ireland
+      </Link>{" "}
+      {new Date().getFullYear()}
+      {"."}
+    </Typography>
+  );
+};
```
Comment (Kivylius) [inline src/pages/Login/LoginPage.tsx]: move to separate coments

File: src/pages/Login/LoginPage.tsx [hunk 3]
```diff
@@ -1,3 +1,111 @@
...
+  const { login } = useAuthContext();
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const credentials: Credentials = { username, password };
+
+  const handleSubmit: FormEventHandler<HTMLFormElement> = async (
+    event: FormEvent
+  ) => {
+    event.preventDefault();
+    await login(credentials);
```
Comment (Kivylius) [inline src/pages/Login/LoginPage.tsx]: you dont need to await this, idealy you have a usestate with loading true: false, to determine it, but its no biggie

File: tsconfig.json [hunk 4]
```diff
@@ -26,5 +26,13 @@
...
     "src"
-  ]
+  ],
+  "baseUrl": "./src",
+  "paths":{
+    "@themes/*": ["themes/*"],
+    "@pages/*": ["pages/*"],
+    "@components/*": ["components/*"],
+    "@context/*": ["context/*"],
+    "@service/*": ["service/*"]
```
Comment (Kivylius) [inline tsconfig.json:36]: that should be by default out of the box. e/g import { a } from "service/a"  ? or you want the @ a the start?

File: src/service/Auth/useAuthService.ts [hunk 5]
```diff
@@ -9,7 +10,7 @@ export type Credentials = {
 export const useAuthService = () => {
   const [token, setToken] = useState("");
```
Comment (Kivylius) [inline src/service/Auth/useAuthService.ts:14]: i would add; 
`const [isLoading, setIsLoading] = useState(false);`
and 
`const [error, setError] = useState(false);` 
 to better display/controll the ui

File: src/service/Auth/useAuthService.ts [hunk 6]
```diff
@@ -9,7 +10,7 @@ export type Credentials = {
...
 
-  const login = async ({ username, password }: Credentials) => {
+  const login: LoginHandler = async ({ username, password }: Credentials) => {
     try {
       const data: { token: string } = await fetchData(
```
Comment (Kivylius) [inline src/service/Auth/useAuthService.ts]: if fix fetchData to be more object based. eg { url: 'a', type: "post"  } ect.

File: src/service/Auth/useAuthService.ts [hunk 7]
```diff
@@ -9,7 +10,7 @@ export type Credentials = {
 export const useAuthService = () => {
   const [token, setToken] = useState("");
 
-  const login = async ({ username, password }: Credentials) => {
+  const login: LoginHandler = async ({ username, password }: Credentials) => {
```
Comment (Kivylius) [inline src/service/Auth/useAuthService.ts:15]: this LoginHandler type is ok, id call it a type e.g LoginHandlerType is all. this make is clear it as a type from first glance

File: src/context/Auth/AuthContext.ts (hunk 1 above)
Comment (fcarelse) [reply src/context/Auth/AuthContext.ts]: Just a placeholder function and in fact does not currently get replaced.
In fact the useAuthService and useAuthContext need to be reworked anyway to get this context to do anything at all.

File: src/pages/Login/LoginPage.tsx (hunk 3 above)
Comment (fcarelse) [reply src/pages/Login/LoginPage.tsx]: done

File: src/service/Auth/useAuthService.ts [hunk 8]
```diff
@@ -9,19 +10,20 @@ export type Credentials = {
...
-        "",
-        { username, password },
-        "post",
-        ""
-      );
+      const data: { token: string } = await fetchData({
+        url: "/user/login",
+        data: { username, password },
+        method: "post",
+        token: "",
```
Comment (Kivylius) [inline src/service/Auth/useAuthService.ts:22]: any reason why you put a token into a function that get a token?

File: src/service/Auth/useAuthService.ts (hunk 7 above)
Comment (fcarelse) [reply src/service/Auth/useAuthService.ts:15]: Done

**Comment by fcarelse** [comment]
Hi

This is passed into the function because login is not the only feature that
will use it. It will need the token to make certain API calls.

Thanks,
 Frank

**Pull request merged**